janus_core/calculations/single_point.py

```diff
@@ -278,60 +315,47 @@ def _get_stress(self) -> MaybeList[ndarray]:
 
     def run(
         self,
-        properties: MaybeSequence[Properties] = (),
-        write_results: bool = False,
+        *,
+        properties: Optional[MaybeSequence[Properties]] = None,
+        write_results: Optional[bool] = None,
         write_kwargs: Optional[OutputKwargs] = None,
     ) -> CalcResults:
         """
         Run single point calculations.
 
         Parameters
         ----------
-        properties : MaybeSequence[str]
-            Physical properties to calculate. If not specified, "energy",
-            "forces", and "stress" will be returned.
+        properties : Optional[MaybeSequence[Properties]]
+            Physical properties to calculate. Default is self.properties.
         write_results : bool
-            True to write out structure with results of calculations. Default is False.
+            True to write out structure with results of calculations. Default is
+            self.write_results.
         write_kwargs : Optional[OutputKwargs],
             Keyword arguments to pass to ase.io.write if saving structure with
-            results of calculations. Default is {}.
+            results of calculations. Default is self.write_kwargs.
 
         Returns
         -------
         CalcResults
-            Dictionary of calculated results.
+            Dictionary of calculated results, with keys from `properties`.
         """
-        results: CalcResults = {}
-        if isinstance(properties, str):
-            properties = [properties]
-
-        for prop in properties:
-            if prop not in get_args(Properties):
-                raise NotImplementedError(
-                    f"Property '{prop}' cannot currently be calculated."
-                )
-
-        # If none specified, get all valid properties
-        if len(properties) == 0:
-            properties = get_args(Properties)
-
-        write_kwargs = write_kwargs if write_kwargs else {}
+        # Parameters can be overwritten, otherwise default to values from instantiation
+        properties = self.properties
+        write_results = write_results if write_results else self.write_results
+        write_kwargs = write_kwargs if write_kwargs else self.write_kwargs
```

Comment:
Do thse need to be `if write_xxx is not None` because if I pass in `False` I presumably want to override what's in `self`?

Reply:
Ah yes, I realised this when I wrote this originally, but forgot and changed it to this form today... Thanks, I'll fix it

Reply:
Do we actually want an alternative to be able to explicitly set `None` too?

I think I saw someone set a global `marker = obj()`, to be a better check against (`if x is not marker` but I'm not sure if there are neater ways.

Reply:
Following discussion the proposal seems to be to remove parameter setting from `run`, simplify the calculation, and require setting of relevant data by the user as property setting or through `__init__`.